Modules/IO/DCMTK/src/itkDCMTKImageIO.cxx

```diff
@@ -28,25 +28,108 @@
 #include "dcmtk/dcmimgle/dcmimage.h"
 #include "dcmtk/dcmjpeg/djdecode.h"
 #include "dcmtk/dcmjpls/djdecode.h"
+#include "dcmtk/dcmdata/dcmetinf.h"
 #include "dcmtk/dcmdata/dcrledrg.h"
 #include "dcmtk/oflog/oflog.h"
 
 namespace
 {
 
 /**
- * Helper function to test for 128 byte dicom preamble
+ * Helper function to test if the file is in standard DICOM file format
+ * (128-byte dicom preamble followed by "DICM" magic word).
  * @param file A stream to test if the file is dicom like
  * @return true if the stream has a dicom preamble
  */
-static bool
-readPreambleDicom(std::ifstream & file)
+bool
+isPreambleDicom(std::ifstream & file)
+{
+  file.seekg(0, std::ios::beg);
+  std::vector<char> buffer(DCM_PreambleLen + DCM_MagicLen);
+  file.read(&buffer[0], buffer.size());
+  if (!file)
+  {
+    return false;
+  }
+  const char * magicReadFromFile = &buffer[DCM_PreambleLen];
+  return strncmp(magicReadFromFile, DCM_Magic, DCM_MagicLen) == 0;
+}
```

Comment:
- Thanks @lassoan Just wondering, how does your pull request relate to #4501 ? Didn't pull request 4501 fix the problem already?
More in detail, isPreambleDicom is just a style improvement to readPreambleDicom, right?

Reply:
Yes, my first commit is just a style improvement (and a little performance improvement) over #4501. There is no functional difference.

#4501 did not fix the issue because the application first tries to read the DICOM image using GDCM and if it fails then it uses DCMTK. However, GDCM did not return with an error but it crashed the application, so the application did not have a chance to use DCMTK.